### ISSUE TITLE
[BUG] With versioning enabled, "Show in Menu" does not work in page tree

### ISSUE THREAD

**Issue by jrief** [django-cms/django-cms#7780]
## Description

On the admin site, the page tree offers a checkbox named "Show in Menu". If djangocms-versioning is enabled, clicking on this checkbox shows an error.

## Steps to reproduce

In the admin open the page tree:

<img width="974" alt="Screenshot 2024-01-24 at 12 19 17" src="https://github.com/django-cms/django-cms/assets/1132472/81a40827-f25a-4ad9-a915-b87a0b303452">

then click on one of the checkboxes in the "Menu" column.

On top of the page tree this error message will appear:

<img width="275" alt="Screenshot 2024-01-24 at 12 20 30" src="https://github.com/django-cms/django-cms/assets/1132472/1d3ce156-0dcc-46da-9757-e6188bedd1b0">

## Expected behaviour

I am well aware that enabling/disabling this setting is not possible with versioning enabled.
However, then this button shall be in read-only mode and not clickable.

## Do you want to help fix this issue?

Yes, if we can agree on this: With djangocms-versioning enabled, the option show/hide in menu can only be changed through the page settings in their detail and not through the page tree.

* [x] Yes, I want to help fix this issue and I will join #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to confirm with the community that a PR is welcome. 
* [ ] No, I only want to report the issue.

**Comment by marksweb** [comment]
Yes that suggestion make sense to me. So hide the column if versioning is installed?

I don't know if there's a better way to do this with the cms config. But I think this is an area we all need to learn more about in order to do things the right way for v4, utilising all the features

@Fabian do you have any thoughts?

**Comment by fsbraun** [comment]
Yes, I would support a change. Probably one that grays out checkboxes for non-draft versions. 

Versioning implements a check framework, so that you can ask a model if you can do something, e.g. `check_publish`. (That might have been a nice thing to have had in the core.) You'll probably have to do `check_modify` for the Version object of the page content in the `has_change_permission` in the `PageContent` admin. This is subclassed in the `cms_config.py` file of versioning.

**Comment by fsbraun** [comment]
I think this needs a **design decision** to move the menu visibility to the page level for django CMS 4.2

**Comment by Will-Hoey** [comment]
@fsbraun @marksweb To add to this. 

The "in menu" can be changed when the Page is in draft, however it will only actually be respected if that version is then published. For example.

1.     Create a Page with "in menu" selected
2.     Mark published. This is visible in the menu as expected
3.     Create new draft and mark as not in menu. Page will remain visible
4.     Promote draft to published. Page is removed from menu

From our perspective, keeping this in the Page admin would be beneficial, however I don't think it should require a new version in order for it to be respected. I'll admit I've not looked at the inner workings here so it might be that it's working that way for a reason!

**Comment by Will-Hoey** [comment]
@fsbraun is the "not planned" tag here accurate? It would be good to get a conversation open around this and see what is possible.

**Comment by fsbraun** [comment]
@Will-Hoey Yeah, the bot marked it. 

The real discussion is if "in menu" should be a Page property, i.e. forced to be identical for all languages, or a PageContent property (as it is now). In the latter case creating a new version is unavoidable.

The argument probably was that some sites only have stripped down translations, i.e. untranslated page contents should not appear in the menu.

I see the usability issue, but have not come around a solution so far.
